src/bin/batch_solve.rs

```diff
@@ -0,0 +1,209 @@
+struct Args {
+    /// Path to configuration file
+    #[arg(required = true)]
+    config: String,
+
+    #[clap(flatten)]
+    boards: Boards,
+
+    /// Directory to output solves to
+    #[arg(short, long, default_value = ".")]
+    dir: String,
+
+    /// Max number of iterations to run
+    #[arg(short = 'n', long, default_value = "1000")]
+    max_iterations: u32,
+
+    /// Default exploitability as ratio of pot. Defaults to 0.02 (2% of pot),
+    /// but for accurate solves we recommend choosing a lower value.
+    #[arg(short = 'e', long, default_value = "0.02")]
+    exploitability: f32,
+
+    /// Overwrite existing sims if a saved sim with the same name exists. By
+    /// default these sims are skipped.
+    #[arg(long, default_value = "false")]
+    overwrite: bool,
+
+    /// Halt the batch solve when encountering a sim with the same name. By
+    /// default these sims are skipped.
+    #[arg(long, default_value = "false")]
+    halt_on_existing: bool,
+
+    /// OOP's range (overwrite the range in the config)
+    #[arg(long)]
+    oop_range: Option<String>,
+
+    /// IP's range (overwrites the range in the config)
+    #[arg(long)]
+    ip_range: Option<String>,
+}
+
+#[derive(Debug, clap::Args)]
+#[group(required = true, multiple = false)]
+struct Boards {
+    /// Path to a file containing a list of boards
+    #[clap(long)]
+    boards_file: Option<String>,
+
+    /// Specify the boards on command line
+    #[clap(long, num_args=1..)]
+    boards: Option<Vec<String>>,
+}
+
+fn main() -> Result<(), String> {
+    let args = Args::parse();
+
+    let boards = if let Some(boards) = args.boards.boards {
+        boards
+    } else {
+        let boards_files = args
+            .boards
+            .boards_file
+            .expect("Must specify boards or boards_file");
+        let boards_contents =
+            std::fs::read_to_string(boards_files).expect("Unable to read boards_file");
+        boards_contents
+            .lines()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>()
+    };
+    let (mut card_config, tree_config) =
+        deserialize_configs_from_file(&args.config).expect("Couldn't deserialize config");
+
+    // Update card_config and tree_config with command-line specified data
+    if let Some(range_string) = args.oop_range {
+        let range_result = range_string.parse::<Range>();
+        if let Ok(range) = range_result {
+            card_config.range[0] = range;
+        } else {
+            println!("Couldn't parse OOP Range \"{}\"", range_string);
+            println!("{}", range_result.unwrap_err());
+        }
+    }
+    if let Some(range_string) = args.ip_range {
+        let range_result = range_string.parse::<Range>();
+        if let Ok(range) = range_result {
+            card_config.range[1] = range;
+        } else {
+            println!("Couldn't parse IP Range \"{}\"", range_string);
+            println!("{}", range_result.unwrap_err());
```

Comment:
I feel like we want to error/exit when we can't parse IP or OOP range.

Reply:
Oh yeah we were supposed to, good catch!